### DISCUSSION TITLE
PreTeXt-CLI-generated websites

### DISCUSSION THREAD

**Discussion by StevenClontz** [PreTeXtBook/pretext-cli#766] [General]
A thread to discuss an experimental feature of PreTeXt-CLi to generate a website for a book.

**Comment by rhinopotamus** [comment]
Some examples of working sites:
- [TBIL.org](https://github.com/TeamBasedInquiryLearning/library/tree/main/site)
- [pretext website](https://github.com/PreTeXtBook/PreTeXt-website/tree/main/site)

Ok, so probably the easiest way to get started fiddling with this experimental feature:
- in your project root directory, create a folder called `/site`
- within that folder, create `site.ptx`
- then make a subfolder called `/pages`
- and then start putting various `.md` files in there for Pelican to eat.

`pretext-cli` parses any xml in `site.ptx` into [Pelican settings](https://docs.getpelican.com/en/latest/settings.html), so that is probably the list of reasonable elements. Apparently you can also specify this kind of thing in [json](https://github.com/PreTeXtBook/PreTeXt-website/blob/main/site/site.json).

The code that does this starts [hereish](https://github.com/PreTeXtBook/pretext-cli/blob/e6cb302a332ef18e3ce4ab8e35eeaf9c3e2cd2d9/pretext/project/__init__.py#L1691).

**Reply by rhinopotamus** [reply]
It would be cool if you could generate `index.html` from markdown.

[The pretext theme](https://github.com/PreTeXtBook/pretext-cli/blob/main/pelican/ptx-theme/templates/index.html) dumps `PTX_SITE_DESCRIPTION` in as raw html. I wonder if there is a way to generate html from markdown before it is dumped in here, maybe through a jinja filter.

I thought about this for a while but didn't see any reasonable way to do it. I'm going to fall back to making a markdown post, having pelican generate it, copy-pasting the html, and then setting the post to hidden.

**Reply by oscarlevin** [reply]
I understand how pelican works better now so I think there are some neat things I could try to get working.  What about a cli function that generated some template files in `/site`?  So if you wanted to get something more complex than the basic theme we generate automatically, you run `pretext init site` or similar.  We could even have `pretext init site --style course|book|personal` that gives you different looks (personal might include a blog, for example).
